kafka/producer.go

```diff
@@ -296,6 +304,13 @@ func (p *Producer) produceBatch(topic string, msgs []*Message, msgFlags int) err
 	cmsgs := make([]C.rd_kafka_message_t, len(msgs))
 	for i, m := range msgs {
 		p.handle.messageToC(m, &cmsgs[i])
+		// Batch production doesn't support a per-message delivery report channel, but it does support
+		// dispatching delivery reports to the main channel. Writing the channel cgoid to _private is
+		// done by set_message_private_cgoid and ensures that eventPoll can find it.
+		if p.handle.fwdDr {
+			cgoid := p.handle.cgoPut(cgoDr{deliveryChan: p.events})
+			C.set_message_private_cgoid(&cmsgs[i], C.uintptr_t(cgoid))
```

Comment:
Each C call is costly. Better to pass the cgoid to messageToC() above.

Reply:
good call. I'll shuffle it around.